.gitlab-ci-check-helm-version-bump.yml

```diff
@@ -0,0 +1,110 @@
+# .gitlab-ci-check-helm-version-bump.yml
+#
+# This gitlab-ci template updates the Helm Chart versions
+# for a given container
+#
+# Requires DOCKER_REPOSITORY variable to be set in the calling Pipeline.
+# Add it to the project in hand through Gitlab's include functionality
+# variables:
+#   DOCKER_REPOSITORY: <Image FQN, i.e mendersoftware/reponame>
+# include:
+#   - project: 'Northern.tech/Mender/mendertesting'
+#     file: '.gitlab-ci-check-helm-version-bump.yml
+# 
+# It requires that the upstream pipeline specifies:
+#
+# - CONTAINER: alvaldi-gui # the container name that later will be translated into
+#                            values.yaml reference
+# - DOCKER_PUBLISH_COMMIT_TAG: ${CI_COMMIT_REF_NAME}_${CI_COMMIT_SHA} 
+# - SYNC_ENVIRONMENT: [staging|prod] # the values-${SYNC_ENVIRONMENT}.yaml to be updated
+# - CHART_DIR: mender # the Helm Chart directory in the helm repo
+# - HELM_PATCH_VERSION: ${CI_PIPELINE_ID} # the Chart.yaml version - use Semver format
+#                                         # prerelases have to be tagged with ${CI_PIPELINE_ID}-staging
+#                                         # to avoid releasing a prerelease to prod (stable semver releases only)
+#
+# The non-production version updates the Chart.yaml with a prerelease version:
+# version: x.y.${CI_PIPELINE_ID}-staging
+# On the downstream, FluxCD is configured to trigger a new Helm deployment every releases and pre-releases on the non-prod envs.
+# At the same way, the production envs are configured to trigger only stable releases.
+
+stages:
+  - version-bump
+
+helm-version-bump:
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "pipeline"
+  stage: version-bump
+  image: registry.gitlab.com/northern.tech/mender/mender-test-containers:aws-k8s-v1-master
+  before_script:
+    - |
+      echo "INFO - setting up git"
+      git config --global user.email "mender@northern.tech"
+      git config --global user.name "Mender Test Bot"
+      export GITHUB_TOKEN=${GITHUB_BOT_TOKEN_REPO_FULL}
+    - |
+      echo "INFO - setting required vars"
+      export CONTAINER_REGISTRY=$(echo ${DOCKER_REPOSITORY%%/*}) # registry.mender.io
+      export CONTAINER_REPOSITORY=$(echo ${DOCKER_REPOSITORY#*/}) # northerntech/alvaldi-gui
+  script:
+    # Adding the GH repo to push back new tags
+    - git remote add github https://${GITHUB_USER}:${GITHUB_TOKEN}@${GITHUB_HELM_REPO}
+    - git fetch github ${MAIN_BRANCH:-master}:overlay-version-bump
+    - git checkout overlay-version-bump
+    # modify this loop to map the source project to an actual values.yaml container definition
+    # e.g.: for the mender-integration/extra/release_tool.py generate-delta-worker becames: generate_delta_worker:
+    # release_tool -m git ${CI_PROJECT_NAME} container
+    - |
+      echo "INFO - mapping source project to values.yaml project:"
+      case $CONTAINER in
+        generate-delta-worker)
+          VALUES_REF="generate_delta_worker"
+          ;;
```

Comment:
Wouldn't this be the same for the catch-all case?
All dashes are replaced with underscores.

Reply:
Not always, some services have different behavior, like the workflow server/worker and some other exception